src/connections.jl

```diff
@@ -319,12 +330,16 @@ Other encodings are not explicitly handled by this package and will probably be
 See also: [`encoding`](@ref), [`reset_encoding!`](@ref)
 """
 function set_encoding!(jl_conn::Connection, encoding::String)
-    status = libpq_c.PQsetClientEncoding(jl_conn.conn, encoding)
+    lock(jl_conn.lock) do
+        status = libpq_c.PQsetClientEncoding(jl_conn.conn, encoding)
 
-    if status == -1
-        error(LOGGER, "libpq could not set the connection's client encoding to $encoding")
-    else
-        jl_conn.encoding = encoding
+        if status == -1
+            error(LOGGER,
+                "libpq could not set the connection's client encoding to $encoding"
+            )
+        else
+            jl_conn.encoding = encoding
+        end
     end
 
     return nothing
```

Comment:
Relevent to this PR but convention would be to have `set` return the thing it was set to, no?

Reply:
Actually, the convention is to return the collection. Assignment expressions add an extra expression to change the return value.

Reply:
Ok